### PR TITLE
Quotas module: Identities and Tiers

### PR DESCRIPTION
**Pull request by filipemmagalhaes**
# Readiness checklist

- [X] I added/updated unit tests.
- [ ] I added/updated integration tests.
- [X] I ensured that the PR title is good enough for the changelog.
- [X] I labeled the PR.

### CODE REVIEW

File: Admin.API/Extensions/QuotasServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -9,14 +9,22 @@ public static class QuotasServiceCollectionExtensions
     public static IServiceCollection AddQuotas(this IServiceCollection services,
         QuotasConfiguration configuration)
     {
+        /*
```
Comment (daniel-almeida-konkconsulting) [inline Admin.API/Extensions/QuotasServiceCollectionExtensions.cs:12]: Please remove this commented out code block

File: Backbone.API/appsettings.override.json [hunk 2]
```diff
@@ -36,6 +36,16 @@
         }
       }
     },
+    "Quotas": {
+      "Application": {},
+      "Infrastructure": {
+        "SqlDatabase": {
+          "Provider": "SqlServer",
+          //"ConnectionString": "User ID=challenges;Password=Passw0rd;Server=postgres;Port=5432;Database=enmeshed;" // postgres
+          "ConnectionString": "Server=ms-sql-server;Database=enmeshed;User Id=challenges;Password=Passw0rd;TrustServerCertificate=True"
```
Comment (daniel-almeida-konkconsulting) [inline Backbone.API/appsettings.override.json:45]: The connection strings are using the 'challenges' user instead of the quotas one.

File: Modules/Quotas/src/Quotas.Application/ApplicationOptions.cs [hunk 3]
```diff
@@ -1,9 +1,8 @@
-﻿using System.ComponentModel.DataAnnotations;
-
-namespace Backbone.Modules.Quotas.Application;
+﻿namespace Backbone.Modules.Quotas.Application;
 
 public class ApplicationOptions
 {
+    /*
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Application/ApplicationOptions.cs:5]: Why was this commented out?

File: Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/IdentityCreatedIntegrationEventHandler.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    public IdentityCreatedIntegrationEventHandler(IIdentitiesRepository identitiesRepository, ILogger<IdentityCreatedIntegrationEventHandler> logger)
+    {
+        _identitiesRepository = identitiesRepository;
+        _logger = logger;
+    }
+
+    public async Task Handle(IdentityCreatedIntegrationEvent integrationEvent)
+    {
+        var identities = new Identity(integrationEvent.Address, integrationEvent.TierId);
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/IdentityCreatedIntegrationEventHandler.cs:19]: variable name should be singular instead of plural

File: Modules/Quotas/src/Quotas.Infrastructure.Database.Postgres/Migrations/20230419134742_Init_Quotas.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#nullable disable
+
+namespace Quotas.Infrastructure.Database.Postgres.Migrations
+{
+    /// <inheritdoc />
+    public partial class Init_Quotas : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Infrastructure.Database.Postgres/Migrations/20230419134742_Init_Quotas.cs:12]: this migration is empty, where are the Tier and Identity tables?

File: Modules/Quotas/src/Quotas.Infrastructure.Database.SqlServer/Migrations/20230419133243_CreatedIdentitiesTiersQuota.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Quotas.Infrastructure.Database.SqlServer.Migrations
+{
+    /// <inheritdoc />
+    public partial class CreatedIdentitiesTiersQuota : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Infrastructure.Database.SqlServer/Migrations/20230419133243_CreatedIdentitiesTiersQuota.cs:13]: Like the Postgres migration, this one is also empty.
In addition to that, both migrations should have similar names.

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/TierQuotaDefinitionsRepository.cs
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/TierQuotaDefinitionsRepository.cs:1]: why was this file deleted?

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/TierQuotaDefinitionsRepository.cs
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/TierQuotaDefinitionsRepository.cs:1]: This is the TierQuotaDefinitionRepository, not the TierRepository.
Why are there changes here? This should be empty for now.

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/HandlerTests.cs
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/HandlerTests.cs:1]: Rename the test class to IdentityCreatedIntegrationEventHandlerTests or change the folder structure.

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/TierQuotaDefinitionsRepository.cs
Comment (filipemmagalhaes) [reply Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/TierQuotaDefinitionsRepository.cs:1]: I removed it to the repository folder, because when I checked in the project the databases were together with the repositories

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/HandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+namespace Backbone.Modules.Quotas.Application.Tests.Tests.Identities
+{
+    public class HandlerTests
+    {
+        private readonly ILogger<IdentityCreatedIntegrationEventHandler> _logger;
+
+        public HandlerTests() { }
+
+        [Fact]
+        public async void Returns_add_when_identities_created()
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/HandlerTests.cs:16]: Rename this test to 'Successfully_creates_identity_after_consuming_integration_event'

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/HandlerTests.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Backbone.Modules.Quotas.Application.Tests.Tests.Tiers;
+public class HandlerTests
+{
+    private readonly ILogger<TierCreatedIntegrationEventHandler> _tiersLogger;
+
+    public HandlerTests() { }
+
+    [Fact]
+    public async void Returns_add_when_tiers_created()
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/HandlerTests.cs:15]: Rename this test to 'Successfully_creates_tier_after_consuming_integration_event'

File: Backbone.API/appsettings.override.json [hunk 9]
```diff
@@ -51,7 +61,8 @@
           "ConnectionString": "Server=ms-sql-server;Database=enmeshed;User Id=devices;Password=Passw0rd;TrustServerCertificate=True"
         },
         "AzureNotificationHub": {
-          "HubName": "nh-dev"
+          "HubName": "nh-dev",
+          "ConnectionString": "teste"
```
Comment (daniel-almeida-konkconsulting) [inline Backbone.API/appsettings.override.json:65]: remove this connectionstring

File: Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/IdentityCreatedIntegrationEventHandler.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        _identitiesRepository = identitiesRepository;
+        _logger = logger;
+    }
+
+    public async Task Handle(IdentityCreatedIntegrationEvent integrationEvent)
+    {
+        var identity = new Identity(integrationEvent.Address, integrationEvent.TierId);
+        await _identitiesRepository.Add(identity, CancellationToken.None);
+
+        _logger.LogTrace($"Successfully created identities. Identity Address: {identity.Address}, Tier ID: {identity.TierId}");
```
Comment (daniel-almeida-konkconsulting) [inline Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/IdentityCreatedIntegrationEventHandler.cs:22]: Change message to singular also -> "Successfully created identity"

**Review by tnotheis — CHANGES_REQUESTED** [review]
Just noticed that I forgot to request changes. Doing that now so that you can re-request review after you're done.

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs [hunk 11]
```diff
@@ -1,11 +1,36 @@
...
+        base.OnModelCreating(builder);
+
+        builder.ApplyConfigurationsFromAssembly(typeof(QuotasDbContext).Assembly);
+        
+        builder.Entity<Identity>(i =>
+        {
+            i.HasOne<Tier>()
+            .WithMany()
+            .HasForeignKey(c => c.TierId);
+        });
```
Comment (tnotheis) [inline Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs:29]: Move this to the IdentityEntityTypeConfiguration

File: Modules/Quotas/src/Quotas.Infrastructure.Database.Postgres/Migrations/20230421155550_Init_Quotas.cs [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+namespace Quotas.Infrastructure.Database.Postgres.Migrations
+{
+    /// <inheritdoc />
+    public partial class Init_Quotas : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Tier",
```
Comment (tnotheis) [inline Modules/Quotas/src/Quotas.Infrastructure.Database.Postgres/Migrations/20230421155550_Init_Quotas.cs:14]: I'm confused. Why is this table called "Tier" instead of "Tiers"? According to the naming conventions, the table is always called like the corresponding DbSet on the DbContext.

That should be fixed. And please don't edit the migrations directly.

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs [hunk 13]
```diff
@@ -1,11 +1,31 @@
...
+        base.OnModelCreating(builder);
+
+        builder.ApplyConfigurationsFromAssembly(typeof(QuotasDbContext).Assembly);
+
+    }
+
+    protected override void ConfigureConventions(ModelConfigurationBuilder configurationBuilder)
+    {
+        base.ConfigureConventions(configurationBuilder);
+    }
```
[the remark this answers is not in the document]
Comment (tnotheis) [reply Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs]: Why do you keep resolving this comment even though you didn't fix the underlying problem?

File: Backbone.API/Program.cs [hunk 14]
```diff
@@ -56,7 +56,7 @@
     .MigrateDbContext<DevicesDbContext>((context, _) => { new DevicesDbContextSeed().SeedAsync(context).Wait(); })
     .MigrateDbContext<FilesDbContext>()
     .MigrateDbContext<RelationshipsDbContext>()
-    .MigrateDbContext<QuotasDbContext>((context, sp) => { new QuotasDbContextSeed(sp.GetRequiredService<DevicesDbContext>()).SeedAsync(context).Wait(); })
+    .MigrateDbContext<QuotasDbContext>()
```
Comment (tnotheis) [inline Backbone.API/Program.cs:59]: Now you don't call your QuotasDbContextSeed anymore

File: Backbone.API/Program.cs [hunk 15]
```diff
@@ -25,9 +25,18 @@
...
+using Backbone.Modules.Files.Infrastructure.Persistence.Database;
+using Backbone.Modules.Relationships.Infrastructure.Persistence.Database;
+using Backbone.Modules.Quotas.Infrastructure.Persistence.Database;
+using Backbone.Modules.Messages.Infrastructure.Persistence.Database;
+using Backbone.Modules.Synchronization.Infrastructure.Persistence.Database;
+using Backbone.Modules.Tokens.Infrastructure.Persistence.Database;
 
 Log.Logger = new LoggerConfiguration()
-    .WriteTo.Console()
+.WriteTo.Console()
```
Comment (tnotheis) [inline Backbone.API/Program.cs:39]: You messed up the formatting here

File: Backbone.API/QuotasDbContextSeed.cs [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+        await SeedTier(context);
+        await AddTierToIdentities(context);
+    }
+
+    private async Task AddTierToIdentities(QuotasDbContext context)
+    {
+        if (await context.Identities.AnyAsync())
+            return;
+
+        foreach (var identityDevices in _devicesDbContext.Identities)
```
Comment (tnotheis) [inline Backbone.API/QuotasDbContextSeed.cs:29]:
```suggestion
        foreach (var sourceIdentity in _devicesDbContext.Identities)
```

File: Backbone.API/QuotasDbContextSeed.cs [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+        await context.SaveChangesAsync();
+    }
+
+    private async Task SeedTier(QuotasDbContext context)
+    {
+        if (await context.Tiers.AnyAsync())
+            return;
+
+        foreach (var tierDevices in _devicesDbContext.Tiers)
```
Comment (tnotheis) [inline Backbone.API/QuotasDbContextSeed.cs:42]:
```suggestion
        foreach (var sourceTier in _devicesDbContext.Tiers)
```

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs [hunk 18]
```diff
@@ -1,11 +1,31 @@
...
 using Enmeshed.BuildingBlocks.Infrastructure.Persistence.Database;
 using Microsoft.EntityFrameworkCore;
 
 namespace Backbone.Modules.Quotas.Infrastructure.Persistence.Database;
 
 public class QuotasDbContext : AbstractDbContextBase
 {
-    public QuotasDbContext(DbContextOptions<QuotasDbContext> options): base(options)
-    { }
+    public QuotasDbContext() { }
```
Comment (tnotheis) [inline Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs:10]: Why the empty constructor?

File: Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs (hunk 11 above)
Comment (tnotheis) [reply Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/QuotasDbContext.cs:29]: You copied it. I said you should **move** it.

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Backbone.Modules.Quotas.Application.Tests.Tests.Identities
+{
+    public class IdentityCreatedIntegrationEventHandlerTests
+    {
+        private readonly ILogger<IdentityCreatedIntegrationEventHandler> _logger;
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs:11]: No need to have this as as field. Just move it to the `CreateHandler` method.

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentityRepository.cs [hunk 20]
```diff
@@ -0,0 +1,16 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Entities;
+
+namespace Backbone.Modules.Quotas.Application.Tests.Tests.Identities;
+public class AddMockIdentityRepository : IIdentitiesRepository
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentityRepository.cs:5]:
```suggestion
public class AddMockIdentitiesRepository : IIdentitiesRepository
```

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+        public IdentityCreatedIntegrationEventHandlerTests() 
+        {
+            _logger = A.Fake< ILogger<IdentityCreatedIntegrationEventHandler>>();
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs:15]:
```suggestion
            _logger = A.Fake<ILogger<IdentityCreatedIntegrationEventHandler>>();
```

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+            var mockIdentityRepository = new AddMockIdentityRepository();
+            var handler = CreateHandler(mockIdentityRepository);
+
+            // Act
+            await handler.Handle(new IdentityCreatedIntegrationEvent() { });
+
+            // Assert
+            mockIdentityRepository.WasCalled.Should().BeTrue();
+            mockIdentityRepository.WasCalledWith.Address.Should().Be(null);
+            mockIdentityRepository.WasCalledWith.TierId.Should().Be(null);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs:31]: Why don't you set the `Address`  and `TierId` properties?! Currently you only test half of the code.

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/AddMockTiersRepository.cs [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+public class AddMockTiersRepository : ITiersRepository
+{
+    public bool WasCalled { get; private set; }
+    public Tier? WasCalledWith { get; private set; }
+
+    public Task Add(Tier tier, CancellationToken cancellationToken)
+    {
+        WasCalled = true;
+        WasCalledWith = tier;
+        return Task.FromResult(WasCalledWith);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/AddMockTiersRepository.cs:14]: Don't return `WasCalledWith`... That doesn't make sense

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentityRepository.cs [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public bool WasCalled { get; private set; }
+    public Identity? WasCalledWith { get; private set; }
+ 
+    public Task Add(Identity identity, CancellationToken cancellationToken)
+    {
+        WasCalled = true;
+        WasCalledWith = identity;
+        return Task.FromResult(WasCalledWith);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentityRepository.cs:14]: Don't return `WasCalledWith`... That doesn't make sense

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+        var mockTierRepository = new AddMockTiersRepository();
+        var handler = CreateHandler(mockTierRepository);
+
+        // Act
+        await handler.Handle(new TierCreatedIntegrationEvent() { });
+
+        // Assert
+        mockTierRepository.WasCalled.Should().BeTrue();
+        mockTierRepository.WasCalledWith.Id.Should().Be(null);
+        mockTierRepository.WasCalledWith.Name.Should().Be(null);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs:30]: Why don't you set the `Id`  and `Name` properties?! Currently you only test half of the code.

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs:31]: Empty blank line

File: Modules/Quotas/test/Quotas.Domain.Tests/TierTests.cs [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+    public class TierTests
+    {
+        [Fact]
+        public void Can_create_tier_with_valid_properties()
+        {
+            var tierId = "1";
+            var tierName = "name";
+            var tier = new Tier(tierId, tierName);
+
+            tier.Id.Should().NotBeNull();
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Domain.Tests/TierTests.cs:16]: You could at least test whether the Id is the one that was passed...

File: Backbone.API/QuotasDbContextSeed.cs
Comment (tnotheis) [inline Backbone.API/QuotasDbContextSeed.cs:1]: You didn't test the seeding, did you?

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentitiesRepository.cs [hunk 28]
```diff
@@ -11,6 +11,6 @@ public Task Add(Identity identity, CancellationToken cancellationToken)
     {
         WasCalled = true;
         WasCalledWith = identity;
-        return Task.FromResult(WasCalledWith);
+        return Task.FromResult(identity);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/AddMockIdentitiesRepository.cs:14]: This makes as much sense as the version you had before...
If the method defines a `Task` as a return value, why would you return a `Task<Identity>`?!?!

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs [hunk 29]
```diff
@@ -8,32 +8,28 @@ namespace Backbone.Modules.Quotas.Application.Tests.Tests.Identities
...
             // Act
-            await handler.Handle(new IdentityCreatedIntegrationEvent() { });
+            await handler.Handle(new IdentityCreatedIntegrationEvent(address, tier));
 
             // Assert
             mockIdentityRepository.WasCalled.Should().BeTrue();
-            mockIdentityRepository.WasCalledWith.Address.Should().Be(null);
-            mockIdentityRepository.WasCalledWith.TierId.Should().Be(null);
+            mockIdentityRepository.WasCalledWith.Address.Should().Be("id12Pbi7CgBHaFxge6uy1h6qUkedjQr8XHfm");
+            mockIdentityRepository.WasCalledWith.TierId.Should().Be("TIRFxoL0U24aUqZDSAWc");
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Identities/IdentityCreatedIntegrationEventHandlerTests.cs:26]: If you define variables, why don't you use them?

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/AddMockTiersRepository.cs [hunk 30]
```diff
@@ -11,6 +11,6 @@ public Task Add(Tier tier, CancellationToken cancellationToken)
     {
         WasCalled = true;
         WasCalledWith = tier;
-        return Task.FromResult(WasCalledWith);
+        return Task.FromResult(tier);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/AddMockTiersRepository.cs:14]: See comment above

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs [hunk 31]
```diff
@@ -7,32 +7,27 @@
...
-        await handler.Handle(new TierCreatedIntegrationEvent() { });
+        await handler.Handle(new TierCreatedIntegrationEvent(id, name));
 
         // Assert
         mockTierRepository.WasCalled.Should().BeTrue();
-        mockTierRepository.WasCalledWith.Id.Should().Be(null);
-        mockTierRepository.WasCalledWith.Name.Should().Be(null);
-        
+        mockTierRepository.WasCalledWith.Id.Should().Be("TIRFxoL0U24aUqZDSAWc");
+        mockTierRepository.WasCalledWith.Name.Should().Be("Basic");
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/TierCreatedIntegrationEventHandlerTests.cs:25]: See comment above

File: Modules/Quotas/test/Quotas.Domain.Tests/TierTests.cs [hunk 32]
```diff
@@ -9,13 +9,12 @@ public class TierTests
...
         public void Can_create_tier_with_valid_properties()
         {
-            var tierId = "1";
-            var tierName = "name";
+            var tierId = "TIREYSCQI6XaMygco7Bw";
+            var tierName = "my-test-tier";
             var tier = new Tier(tierId, tierName);
 
-            tier.Id.Should().NotBeNull();
+            tier.Id.Should().Be("TIREYSCQI6XaMygco7Bw");
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Domain.Tests/TierTests.cs:16]: See comment above

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]
